internal/config/config.go

```diff
@@ -1274,6 +1285,13 @@ func buildSerializer(name string, tbl *ast.Table) (serializers.Serializer, error
 		}
 	}
 
+	// if the output data format is JSON, then set the timestamp
+	// units for the serializer configuration to the units that were
+	// defined in the AgentConfig
+	if c.DataFormat == "json" {
+		c.TimestampUnits = agentConfig.JsonTimestampUnits
+	}
+
```

Comment:
The `json_timestamp_units` (as I defined it in this PR) is actually part of the agent configuration, not part of the configuration for the output filters (this was in response to my previous PRs on this topic, where the feedback I got was that I shouldn't have to touch any of the outputs; all of the changes should be made in the serializer itself).  Since it's not part of the output filter, it's not available in the `pluginSubTable` that is passed into this method from the `LoadConfig` method (through the `addOutput` method).  Is there a way to get to the `AgentConfig` from within this method?  If so, I'd be more than happy to make the change, but as things stand if I modify this to look more like the code that parses the `prefix` and `template` values for the graphite output filter, the end result is that the TimestampUnits are never set for the serializer configuration (since the `json_timestamp_units` are not part of the `pluginSubTable` being parsed here).  Does that make sense?